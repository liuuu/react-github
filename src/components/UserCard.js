import React from "react";
import { Col, Card, Avatar } from "antd";
import { Link } from "react-router-dom";

const UserCard = props => {
  return (
    <div className="usercard">
      <Col span={8} style={props.style} style={{ padding: "20px" }}>
        <Link to={`/user/${props.owner.login}`} style={{ color: "black" }}>
          <Card>
            <div className="popular-rank">
              #{parseInt(`${props.index}`) + 1}
            </div>
            <ul className="space-list-item">
              <li>
                <Avatar
                  className="avatar"
                  src={props.owner.avatar_url}
                  alt={props.owner.login}
                />
              </li>
              <li><p>{props.name}</p></li>
              <li>@{props.owner.login}</li>
              <li>Stars: {props.stargazers_count}</li>
            </ul>

          </Card>
        </Link>
      </Col>
    </div>
  );
};
UserCard.defaultProps = {
  owner: {
    login: "testing"
  }
};
export default UserCard;
