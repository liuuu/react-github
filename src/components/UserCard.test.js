import React from "react";
import UserCard from "./UserCard";
// import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { shallow, mount, render } from "enzyme";
import { Col } from "antd";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";

const props = {
  title: "string",
  description: "string",
  year: "string",
  imdbID: "string",
  poster: "string",
  trailer: "string",
  name: "string",
  id: "string",
  style: "mixed",
  owner: {
    login: "string",
    avatar_url: "string"
  },
  key: "string",
  stargazers_count: "string"
};

it("render UserCard correctly", () => {
  const component = render(
    <MemoryRouter>
      <UserCard {...props} />
    </MemoryRouter>
  );
  expect(toJson(component)).toMatchSnapshot();
});
//
// it("render UserCard correctly", () => {
//   const component = render(<UserCard {...props} />);
//   expect(toJson(component)).toMatchSnapshot();
// });
