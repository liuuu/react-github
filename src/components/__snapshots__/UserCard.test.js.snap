// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render UserCard correctly 1`] = `
<div
  class="usercard"
>
  <div
    class="ant-col-8"
    style="padding:20px;"
  >
    <a
      href="/user/string"
      style="color:black;"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="popular-rank"
          >
            #NaN
          </div>
          <ul
            class="space-list-item"
          >
            <li>
              <span
                alt="string"
                class="ant-avatar avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="string"
                />
              </span>
            </li>
            <li>
              <p>
                string
              </p>
            </li>
            <li>
              @string
            </li>
            <li>
              Stars: string
            </li>
          </ul>
        </div>
      </div>
    </a>
  </div>
</div>
`;
